v2/pkg/engine/datasource/httpclient/nethttpclient.go

```diff
@@ -190,8 +194,10 @@ func makeHTTPRequest(client *http.Client, ctx context.Context, url, method, head
 		return err
 	}
 	defer response.Body.Close()
+	request.Header = redactHeaders(request.Header)
```

Comment:
This looks wrong to me because we don't want to modify the original headers. We only want to redact them for analytics.

Reply:
I could argue that we should modify the headers, because clients shouldn't be accessing sensitive information like `authorization`/`cookie` headers. I can also expose the `redactHeaders` function so that we can utilize it in our hook implementation